### DISCUSSION TITLE
Problem with copying image with mask

### DISCUSSION THREAD

**Discussion by vavilon224** [fabricjs/fabric.js#8924] [General]
export const createImageMask = async (canvas, points, url) => {
  if (url) {
    const image = new Image();
    image.src = url;
    image.crossOrigin = 'anonymous';

    image.onload = function () {
      const img = new fabric.Image(image);

      const svgMask = new fabric.Polygon(points, {
        originX: 'center',
        originY: 'center',
        strokeWidth: 0,
        layersType: 'shape',
        left: 0,
        top: 0,
        strokeLineJoin: 'bevil',
      });

      svgMask.scaleToWidth(300);
      svgMask.scaleToHeight(300);

      img.set({
        imageFormat: 'image',
        navigationType: 'images',
        originX: 'center',
        originY: 'center',
        flipX: false,
        flipY: false,
        layersType: 'image',

        imageFilters: {},
      });

      img.scaleToWidth(300);
      img.scaleToHeight(300);

      if (points) {
        img.setCoords();

        const group = new fabric.Group([img], {
          clipPath: svgMask,
          layout: 'clip-path',
          imageFormat: 'image',
          navigationType: 'images',
          layersType: 'image',
        });
        group.scaleToWidth(300);
        group.scaleToHeight(300);
        group.center().setCoords();
        centerObject(group, canvas);
        canvas.add(group).setActiveObject(group).renderAll();
      } else {
        img.center().setCoords();
        centerObject(img, canvas);
        canvas.add(img).setActiveObject(img).renderAll();
      }
    };
  }
};

**Comment by vavilon224** [comment]
Here is my code for creating an image with a mask.
But when copying, the pictures disappear, but as soon as the target drops, they appear

@ShaMan123  @melchiar Guys, sorry to bother you. I just tried everything I can

https://github.com/fabricjs/fabric.js/discussions/8925

**Comment by vavilon224** [comment]
![2](https://github.com/fabricjs/fabric.js/assets/87662684/703211f9-7dd7-4ec2-938c-027d7fc42877)
![1](https://github.com/fabricjs/fabric.js/assets/87662684/d070ccaa-c2d3-4a48-b0b8-c2656f420d11)

**Comment by vavilon224** [comment]
![3](https://github.com/fabricjs/fabric.js/assets/87662684/c3fa5869-b4f2-41c2-8281-62ee0c341660)

**Comment by vavilon224** [comment]
const copyPastHandler = (eventData, transform) => {

    const selectedObject = transform.target;

      selectedObject.clone(function (cloned) {
        let localIndex = parseInt(loadFromLocalStorage(`index-${canvas.workSide}-${productId}`));

        cloned.set({
          left: cloned.left + 10,
          top: cloned.top + 10,
          objectId: nanoid(),
          objectName: `object №${localIndex + 1 || 1}`,
          evented: true,
          zIndex: localIndex + 1 || 1,
        });

        if (cloned.type === 'activeSelection') {
          cloned.canvas = canvas;
          cloned.forEachObject(function (obj) {
            canvas.add(obj);
          });
          cloned.setCoords();
        } else {
          canvas.add(cloned);
        }

        cloned.top += 10;
        cloned.left += 10;
        canvas.setActiveObject(cloned);
        canvas.requestRenderAll();
      });
  };



And if I use normal copying, I get an error:

Uncaught TypeError: this._cacheContext.clearRect is not a function

https://github.com/fabricjs/fabric.js/discussions/8925

**Comment by ShaMan123** [comment]
If you want my help make a repro in v6.

**Reply by ShaMan123** [reply]
Anything other than that is a waste of time IMO
